pyeudiw/storage/mongo_storage.py

```diff
@@ -225,34 +224,58 @@ def _add_entry(
         db_collection = getattr(self, collection)
         db_collection.insert_one(attestation)
         return entity_id
+    
+    def _update_attestation_metadata(self, ta: dict, attestation: list[str], exp: datetime, trust_type: TrustType):
+        trust_name = "x509" if trust_type == TrustType.X509 else "federation"
+        attestation_name = "x5c" if trust_type == TrustType.X509 else "chain"
+
+        trust_entity = ta.get(trust_name, None) or {}
+
+        trust_entity[attestation_name] = attestation
+        trust_entity["exp"] = exp
+
+        ta[trust_name] = trust_entity
+
+        return ta
+    
+    def _update_anchor_metadata(self, ta: dict, attestation: list[str], exp: datetime, trust_type: TrustType):
+        trust_name = "x509" if trust_type == TrustType.X509 else "federation"
+        attestation_name = "pem" if trust_type == TrustType.X509 else "entity_configuration"
+
+        trust_entity = ta.get(trust_name, None) or {}
 
-    def add_trust_attestation(self, entity_id: str, attestation: list[str], exp: datetime):
+        trust_entity[attestation_name] = attestation
+        trust_entity["exp"] = exp
+
+        ta[trust_name] = trust_entity
+
+        return ta
+
+    def add_trust_attestation(self, entity_id: str, attestation: list[str], exp: datetime, trust_type: TrustType):
         entity = {
             "entity_id": entity_id,
-            "federation": {
-                "chain": attestation,
-                "exp": exp
-            },
+            "federation": {},
             "x509": {}
         }
 
+        ta = self._update_attestation_metadata(entity, attestation, exp, trust_type)
+
         self._add_entry(
-            "trust_attestations", entity_id, entity, exp
+            "trust_attestations", entity_id, ta, exp
```

Comment:
a trust attestation (truist chain or x509 certificate chain) is related to an entity

a TA is a trust anchor that's an entity but a specific entity type

then the attestation is related to any kind of entity, not just the TA